File: ruby/gilded_rose.rb
require File.join(File.dirname(__FILE__), 'item')
require File.join(File.dirname(__FILE__), 'gilded_rose_item')

class GildedRose

  def initialize(items)
    @items = items
  end

  def update_quality()
    @items.each do |item|
      gilded_rose_item = GildedRoseItem.new(item).update
    end
  end
end

File: csharpcore/GildedRose/GildedRose.cs
using System.Collections.Generic;

namespace GildedRoseKata;

public class GildedRose
{
  IList<Item> Items;

  public GildedRose(IList<Item> Items)
  {
    this.Items = Items;
  }

  public void UpdateQuality()
  {
    foreach (Item item in Items)
    {
      GildedRoseItem gildedRoseItem = new GildedRoseItem(item);
      
      gildedRoseItem.Update();
    }
  }
}
